data/pigui/libs/modal-win.lua

```diff
@@ -0,0 +1,81 @@
+-- Copyright © 2008-2020 Pioneer Developers. See AUTHORS.txt for details
+-- Licensed under the terms of the GPL v3. See licenses/GPL-3.txt
+
+local ui = import 'pigui/pigui.lua'
+
+local modals = {}
+local modalStack = {}
+
+local ModalWindow = {}
+
+function ModalWindow.New(name, innerHandler, outerHandler, flags)
+	local self = {
+		name = name,
+		flags = flags or {"NoTitleBar", "NoResize", "AlwaysAutoResize", "NoMove"},
+		stackIdx = -1,
+		isOpen = false,
+		innerHandler = innerHandler,
+		outerHandler = outerHandler,
```

Comment:
I'd also recommend that you set `outerHandler = outerHandler or function(drawFn) drawFn() end` instead of deferring the check until the end of the table creation. You save about five lines that way :D

Reply:
I don't know what I was thinking :D